--- droidbox.py	2015-04-30 15:59:27.114049745 +0200
+++ droidbox.py	2015-08-25 13:04:27.414127662 +0200
@@ -35,7 +35,7 @@
 
 from threading import Thread
 from xml.dom import minidom
-from subprocess import call, PIPE, Popen
+from subprocess import call, PIPE, STDOUT, Popen
 from utils import AXMLPrinter
 import hashlib
 from pylab import *
@@ -312,7 +312,7 @@
 		sys.exit(1)
 
 	#Execute the application
-	ret = call(['monkeyrunner', 'monkeyrunner.py', apkName, packageName, mainActivity], stderr=PIPE, cwd=os.path.dirname(os.path.realpath(__file__)))
+	ret = call(['monkeyrunner', 'monkeyrunner.py', apkName, packageName, mainActivity], stderr=STDOUT, cwd=os.path.dirname(os.path.realpath(__file__)))
 
 	if (ret == 1):
 		print("Failed to execute the application.")
@@ -359,7 +359,12 @@
 	    signal.alarm(duration)
 
 	#Collect DroidBox logs
+        lastScreenshot = 0
 	while 1:
+            if (time.time() - lastScreenshot) >=5:
+                #Take Screenshots every 5 seconds.
+                os.system("adb shell screencap -p | sed 's/\r$//' > /samples/out/screen_$(date +%Y-%m-%d_%H%M%S).png")
+                lastScreenshot = time.time()
 	    try:
 		logcatInput = adb.stdout.readline() 
 		if not logcatInput:
@@ -507,6 +512,8 @@
 	output["apkName"] = apkName
 
 	print(json.dumps(output))
+	with open("/samples/out/analysis.json","w") as jsonfile:
+	    jsonfile.write(json.dumps(output,sort_keys=True, indent=4))
 	sys.exit(0)
 
 if __name__ == "__main__":
